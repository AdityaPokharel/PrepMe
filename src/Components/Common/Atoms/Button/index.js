import React from 'react';
import PropTypes from 'prop-types';
import {Text, TouchableOpacity, StyleSheet} from 'react-native';

export const Button = ({label, onPress}) => {
  const styles = styleSheet();
  return (
    <TouchableOpacity
      activeOpacity={0.9}
      style={styles.wrapper}
      onPress={onPress}>
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styleSheet = theme =>
  StyleSheet.create({
    wrapper: {
      width: '100%',
      height: 48,
      backgroundColor: '#4d13da',
      justifyContent: 'center',
      borderRadius: 5,
    },
    text: {
      textAlign: 'center',
      color: 'white',
      fontWeight: 'bold',
    },
  });
