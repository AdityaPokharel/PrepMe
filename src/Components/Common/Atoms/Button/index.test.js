import React from 'react';
import {TouchableOpacity} from 'react-native';
import renderer from 'react-test-renderer';
import {fireEvent, render} from 'react-native-testing-library';
import Button from '.';

describe('button tests', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Button onPress={() => {}} label="test" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render button label', () => {
    const label = 'test label';
    const {getByText} = render(<Button onPress={() => {}} label={label} />);
    getByText(label);
  });

  it('should call onPress on button press', () => {
    const onPress = jest.fn();
    const {getByType} = render(<Button onPress={onPress} label="test label" />);
    const element = getByType(TouchableOpacity);
    fireEvent.press(element);
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
