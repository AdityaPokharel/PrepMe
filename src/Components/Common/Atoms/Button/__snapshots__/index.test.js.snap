// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4d13da",
      "borderRadius": 5,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
